okhttp/src/commonMain/kotlin/okhttp3/internal/-HttpUrlCommon.kt

```diff
@@ -0,0 +1,904 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal
+
+import kotlin.jvm.JvmStatic
+import okhttp3.HttpUrl
+import okhttp3.internal.HttpUrlCommon.canonicalize
+import okhttp3.internal.HttpUrlCommon.writePercentDecoded
+import okio.Buffer
+
+internal expect object HttpUrlCommon {
+  internal fun Buffer.writePercentDecoded(
```

Comment:
I wouldn’t have expected this to need an `expect fun` ?

Reply:
Called by `String.percentDecode` in common code.